src/dataflow/src/source/s3.rs

```diff
@@ -285,7 +291,10 @@ async fn download_object(
                 let activate = !buf.is_empty();
                 let mut lines = 0;
                 for line in buf.split(|b| *b == b'\n').map(|s| s.to_vec()) {
-                    if let Err(e) = tx.send(Ok(line)) {
+                    if let Err(e) = tx.send(Ok(InternalMessage {
+                        bucket: bucket.clone(),
```

Comment:
Can we avoid this clone? (For example, one way to do it would be to have the overall source hold an array of bucket strings, and have this task send over the index of the bucket in that array, rather than the name itself).

Or do you think that would be a premature optimization?

Reply:
the issue is that it would later get cloned later for the PartitionID anyway as part of `NextMessage::Ready` line 361 in this diff.

I'll try and optimize it out into some Rc-ish thing, but I _think_ it'll get sent into dataflow, where we require a String anyway.

Reply:
A clone on line 361 would not be as bad, since that will be running on the dataflow worker, so at least the cloned data doesn't need to be sent to a different core.

Anyway, don't spend too much time on it imo, a `// TODO[perf] can we use some kind of string pool to avoid this clone?` should be fine.